### ISSUE TITLE
Janus Video room plugin allows rooms with negative numbers

### ISSUE THREAD

**Issue by Computician** [meetecho/janus-gateway#241]
I was able to create a room with an ID of -1 and that caused an underflow for the actual room id...which turned into 18446744073709551615...

I am away from my build environment for janus or I would push a patch myself.

**Comment by lminiero** [comment]
Good catch, I'll fix that tomorrow when I get back to the lab. I guess the same applies to some other numeric identifiers we make use of throughout core and other plugins.

**Comment by lminiero** [comment]
Sorry, the tomorrow turned out to be a bit later than that... I got stuck in some other activities and forgot about this.

I just tried to reproduce it myself and I'm not getting the same behaviour: the negative number is returned exactly as it is, both when creating a new Janus session -1, and when creating a new videoroom -1 (which makes sense, as they use the same approach and both are guint64 values). Not sure what's causing this in your case, it may be a host issue. The problem remains anyway, as negative IDs don't make much sense, and so we should check them before using them.

I'll try and work on a patch soon, which might take a while since similar IDs are used all over the place.
